### ISSUE TITLE
aedatastore cannot bind embedded pointer of struct

### ISSUE THREAD

**Issue by micnncim** [mercari/datastore#52]
Thank you for awesome package.

I tried to get embedded struct but failed. Show the cases below (ignore error).
The difference of the cases is `Bar` is `struct` or `pointer of struct`.
Is this a bug?

### failure case

The embedded `Bar` is null.

```go
type Foo struct {
	_kind string `boom:"kind"`
	ID string `boom:"id"`
	Bar *Bar `datastore:",flatten"`
}

type Bar struct {
	Name string
}

func get() {
	client, _ := aedatastore.FromContext(ctx)
	bm := boom.FromClient(ctx, client)
	foo := Foo{ID:"1"}
	bm.Get(&foo)
        b, _ := json.Marshal(foo)
        fmt.Print(b)
        // => {"ID":"1","Bar":null}
}
```

## success case

```go
type Foo struct {
	_kind string `boom:"kind"`
	ID string `boom:"id"`
	Bar Bar `datastore:",flatten"`
}

type Bar struct {
	Name string
}

func get() {
	client, _ := aedatastore.FromContext(ctx)
	bm := boom.FromClient(ctx, client)
	foo := Foo{ID:"1"}
	bm.Get(&foo)
        b, _ := json.Marshal(foo)
        fmt.Print(b)
        // => {"ID":"1","Bar":{"Name":"bar1"}}
}
```

**Comment by micnncim** [comment]
Thank you for your very helpful answer.
The test really makes sense. I've understood how it works.
I'll watch the issues and discussion.

**Comment by vvakame** [comment]
https://github.com/googleapis/google-cloud-go/issues/1296#issuecomment-467181713 👀

**Comment by vvakame** [comment]
root issue is solved at v0.37.0 https://github.com/googleapis/google-cloud-go/releases

**Comment by micnncim** [comment]
I sincerely appreciate your reporting!
